core/plugin_manager.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Manager for all items registered with the workspace.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+
+'use strict';
+
+goog.provide('Blockly.PluginManager');
+
+goog.require('Blockly.IPlugin');
+
+
+/**
+ * Manager for all items registered with the workspace.
+ * @constructor
+ */
+Blockly.PluginManager = function() {
```

Comment:
I'd like to rename this and `IPlugin` (but I don't yet have a good alternative in mind) away from using the word "plugin". 
We are currently using plugins to refer to objects/overrides that developers can add on top of core Blockly (such as where we pass in inject plugin options classes to override), but this manager is also used to track core Blockly elements (like trashcan and zoom).

Reply:
Any thoughts on FeatureManager?

Reply:
I'm not sold on `FeatureManager` because I think it's not specific enough. Feature is a very encompassing term, and there's plenty of features that this doesn't manage.
I think it might get easier to find a name after we try to use this a bit and get a better idea of what things we would add to the manager, other than `IPositionable`.